sample/AspNetFullFrameworkSampleApp/Data/SampleDataDbContext.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Data.Common;
+using System.Data.Entity;
+using System.Data.SQLite;
+using System.Threading;
+using SQLite.CodeFirst;
+
+namespace AspNetFullFrameworkSampleApp.Data
+{
+	internal class SampleDataDbContext : DbContext
+	{
+		private static readonly InitOnceHelper InitOnceHelperInstance = new InitOnceHelper();
+		private static readonly Lazy<DbConnection> Connection = new Lazy<DbConnection>(CreateDbConnection);
+
+		public SampleDataDbContext(object attachedState = null)
+			: base(Connection.Value, /* contextOwnsConnection */ false)
+		{
+			Configure();
+			InitOnceHelperInstance.Init(() => { Database.Initialize( /* force: */ true); });
+			AttachedState = attachedState;
+		}
+
+		internal object AttachedState { get; }
```

Comment:
I see you use this `AttachedState` in the `ConcurrentDbTest` test, but after some time I gave up on understanding what this is. Maybe a short comment on this would be useful.

Reply:
Done.